### PR TITLE
17wf/polls

### PR DESCRIPTION
**Pull request by Piwakk**

### CODE REVIEW

File: polls/models.py [hunk 1]
```diff
@@ -44,26 +47,39 @@ class Poll(models.Model):
...
         """
-        Return True if the poll is has been published in the past, that is publication_date_time <= now(),
-        and its state is ACCEPTED.
+        Return True if the poll is has been published in the past, that is publication_date <= now(), and its state is
+        ACCEPTED.
         """
         td = date.today()
         if self.publication_date is None or self.state != 'ACCEPTED':
             return False
         return self.publication_date <= td
```
Comment (abocquet) [inline polls/models.py]: No need to declare a tmp variable that will be read once, not modified and short to write

File: polls/permissions.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+            return (request.user.is_staff or  # Administrators can always retrieve.
+                    request.user == poll.user or  # Authors can always retrieve.
+                    poll.has_been_published)  # Published polls are public.
+        elif request.method in ('PATCH',):
+            self.message = 'You are not allowed to update this poll.'
+
+            return (request.user.is_staff or  # Administrators can always update.
+                    (request.user == poll.user and
+                     not poll.state == 'ACCEPTED'))  # Authors can only update if the poll is not accepted.
```
Comment (abocquet) [inline polls/permissions.py]: Are `ACCEPTED` and `PENDING` the two only states of a poll ? If not, this permission might be leaky

Comment (Piwakk) [reply polls/permissions.py]: Three states: `REVIEWING` (not `PENDING`), `ACCEPTED` and `REJECTED`.
So yes, let's forbid the update if the state is `REJECTED`.

Comment (Piwakk) [reply polls/permissions.py]: Three states: `REVIEWING` (not `PENDING`), `ACCEPTED` and `REJECTED`.
So yes, let's forbid the update if the state is `REJECTED`.

**Pull request merged**